plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any, List, MutableMapping, Optional
+
+import optuna
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.override_parser.types import (
+    ChoiceSweep,
+    IntervalSweep,
+    Override,
+    RangeSweep,
+    Transformer,
+)
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, OmegaConf
+from optuna.distributions import (
+    CategoricalDistribution,
+    DiscreteUniformDistribution,
+    IntLogUniformDistribution,
+    IntUniformDistribution,
+    LogUniformDistribution,
+    UniformDistribution,
+)
+from optuna.samplers import CmaEsSampler, RandomSampler, TPESampler
+
+from .config import DistributionConfig, OptunaConfig
+
+log = logging.getLogger(__name__)
+
+
+def create_optuna_distribution_from_config(config: MutableMapping[str, Any]) -> Any:
+    param = DistributionConfig(**config)
+    if param.type == "categorical":
+        assert param.choices is not None
+        return CategoricalDistribution(param.choices)
+    if param.type == "int":
+        assert param.low is not None
+        assert param.high is not None
+        if param.log:
+            return IntLogUniformDistribution(int(param.low), int(param.high))
+        return IntUniformDistribution(
+            int(param.low), int(param.high), step=int(param.step)
+        )
+    if param.type == "float":
+        assert param.low is not None
+        assert param.high is not None
+        if param.log:
+            return LogUniformDistribution(param.low, param.high)
+        if param.step != 1:
```

Comment:
If this `if condition` added because `DiscreteUniformDistribution` uses 1 as the step size by default?

Reply:
Ah, it was not expected behavior. Optuna sweeper should be able to create `DiscreteUniformDistribution` with `step=1`. I fixed the logic in https://github.com/facebookresearch/hydra/pull/1132/commits/0c21081dad8b5fec0f40695c705dd580f9aa0aad.